### ISSUE TITLE
combineBCR() Error in rbind.data.frame(out, uni_IG) : 找不到对象'uni_IG'

### ISSUE THREAD

**Issue by shijianasdf** [ncborcherding/scRepertoire#147]
When I run combineBCR() as follow:
> ```r
>S2 <- read.csv("/Volumes/HDD3/tianjin_ICC/P5_665746/BCR/outs/filtered_contig_annotations.csv")
>contig_list <- list(S2)
>combined <- combineBCR(contig_list,samples = c("P5"), ID=c("665746"))
>```
then I got error as follow:
> ```r
>INFO [2021-09-27 08:40:40] ::plot_cnv:Depending on the size of the matrix this may take a moment.
>INFO [2021-09-27 08:41:23] plot_cnv(): auto thresholding at: (0.834789 , 1.165211)
>Error in rep(contig_name, contig_tbl[contig_name]) : 'times'参数不对
> ```

**Comment by shijianasdf** [comment]
> Thanks for reaching out - what version of scRepertoire are you using?

scRepertoire_1.4.0

**Comment by ncborcherding** [comment]
Hey Biowalker, 

None of these messages are being generated from scRepertoire, it looks like they are from the package [infercnv](https://github.com/broadinstitute/infercnv).  I am not exactly sure why they are appearing when you are calling ```combineBCR()``` but I would check to see if the **combined** object is created? 

I am going to close this issue as the messages are not from scRepertoire, but feel free to comment below and we can continue to troubleshoot if needed. 

Nick

**Comment by s2hui** [comment]
Hello,
I am experiencing the same error but I am not using inferCV:
```
contig_list = list()
for(i in 1:length(EXT_IDS)) {
+    name = EXT_IDS[i]
+    csv = read.csv(paste0("/data/raw/vdj/",name,".vdj/filtered_contig_annotations.csv"), stringsAsFactors = F)
+    contig_list[[i]] = csv
+ }
combined <- combineBCR(contig_list, samples = IDS, ID = REGION_NUM,removeNA = TRUE,removeMulti=TRUE)
Error in rbind.data.frame(out, uni_IG) : object 'uni_IG' not found
combined
Error: object 'combined' not found
```
I wonder what the problem might be?  I am able to run `combineTCR()` on the same contig_list with no problems.

Version:
`scRepertoire_1.1.4`

Thanks for your help,
@s2hui

**Comment by s2hui** [comment]
Hello, Yes I think that worked.  Thanks for your quick reply!
@s2hui

**Comment by bobermayer** [comment]
Hi,
I'm also getting this error when subsetting a `filtered_contig_annotations.csv` to barcodes from a particular sample in a multiplexed pool (using v1.6.0):
```
> tmp <- combineBCR(bcr[bcr$barcode %in% bcs,],samples=c('tmp'),ID=c('tmp'))
Error in rbind.data.frame(out, uni_IG) : object 'uni_IG' not found
```
I don't see an error if I don't subset. curiously, using scRepertoire v1.5.3, I get
```
> tmp <- combineBCR(bcr[bcr$barcode %in% bcs,],samples=c('tmp'),ID=c('tmp'))
Error in `$<-.data.frame`(`*tmp*`, "cluster", value = "IGH:LD.") : 
  replacement has 1 row, data has 0
```

any ideas? maybe related to #216?

thanks!

**Comment by ncborcherding** [comment]
Hey Benedikt, 

Thanks for the excellent rundown of the problem. The issue stems from the previous versions (both 1.5.3 and 1.7.0) assuming all samples would have edit distance-based clonotypes called during the ```combineBCR()``` process. This has been fixed in the most recent dev version (v1.7.3) - bunch of refinements for loading contigs and BCR work and obviously I was too confident in my commit and broke things. Let me run a quick QC on things and I will update in < 1 hour. 

Nick

**Comment by bobermayer** [comment]
hi Nick,
thanks for the update, it solved this problem!
Benedikt
